src/pages/Content/index.ts

```diff
@@ -1,2 +1,58 @@
-//eslint-disable-next-line no-console
-console.log('Content script works!')
+import { KITSPACE_PROCESSOR_API_KEY } from 'secrets'
+
+const processorDomain = 'https://processor.master.staging.kitspace.dev'
+
+const headers = {
+  Authorization: `Bearer ${KITSPACE_PROCESSOR_API_KEY}`,
+  Accept: 'application/json',
+}
+
+async function replaceKicadPcbWithSvg() {
+  const rawButton = document.getElementById('raw-url')
+  if (rawButton == null) {
+    console.error('Kitspace WebExtension could not find "Raw" link')
+    return
+  }
+  const rawUrl = rawButton.getAttribute('href')
+  if (rawUrl == null) {
+    console.error('Kitspace WebExtension could not get `href` from "Raw" link')
+    return
+  }
+
+  const blob = await fetch(rawUrl).then(r => r.blob())
+
+  const formData = new FormData()
+  formData.append('upload', blob, 'x.kicad_pcb')
+
+  const { id } = await fetch(processorDomain + '/process-file', {
+    method: 'POST',
+    body: formData,
+    mode: 'cors',
+    headers,
+  }).then(r => r.json())
+
+  const statusUrl = `${processorDomain}/processed/status/${id}/images/layout.svg`
+
+  let r = await fetch(statusUrl).then(r => r.json())
+
+  while (r.status !== 'done' && r.status !== 'failed') {
+    r = await fetch(statusUrl).then(r => r.json())
+  }
```

Comment:
We definitely don't want to loop like I do here. @AbdulrhmnGhanem has been using [exponential-backoff](https://github.com/coveooss/exponential-backoff) with success in a few places.

Reply:
[Long polling](https://javascript.info/long-polling) may be a better approach to this in general (but will also require API changes if I understand it correctly).